### PR TITLE
PLAT-9431 Add gift vouchers Shortcode

### PR DESCRIPTION
**Pull request by jadkhateradmin**
https://administrate.atlassian.net/browse/PLAT-9431

### CODE REVIEW

File: src/Shortcodes/Shortcode.php [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+         * */
+        public static function instance()
+        {
+            if (!isset(self::$instance)) {
+                $className = __CLASS__;
+                self::$instance = new $className;
+            }
+
+            return self::$instance;
+        }
```
Comment (alihabib-administrate) [inline src/Shortcodes/Shortcode.php:47]: do we really need it ? can it be replaced with a constructor ?

Comment (alihabib-administrate) [inline src/Shortcodes/Shortcode.php:47]: do we really need it ? can it be replaced with a constructor ?

Comment (jadkhateradmin) [reply src/Shortcodes/Shortcode.php:47]: @alihabib-administrate creating the instance of this class is needed to fire up the creation of the filters / action / shortcodes wordpress hooks.
We do have a __construct method for this class also that is called upon instance creation by `new $className`
In Main.php we are instantiating the shortcode class.

**Review by alihabib-administrate — APPROVED** [review]
LGTM

**Review by omaya-adm — APPROVED** [review]

**Pull request merged**